modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessPlugin.java

```diff
@@ -34,7 +35,7 @@
 /**
  * Registers Painless as a plugin.
  */
-public final class PainlessPlugin extends Plugin implements ScriptPlugin {
+public final class PainlessPlugin extends Plugin implements ScriptPlugin, ExtensiblePlugin {
```

Comment:
shouldn't this implement `reloadSPI` ?

Reply:
Painless doesn't have any SPI interfaces yet. This is just the scaffolding to add it (as an example of how SPI works in general for this PR).